sdk/nodejs/src/request.ts

```diff
@@ -43,87 +14,119 @@ const TRACK_PATH = '/v1/track'
 const BUCKETING_URL = `${BUCKETING_BASE}${HOST}`
 const EDGE_DB_QUERY_PARAM = '?enableEdgeDB='
 
+export class ResponseError extends Error {
+    constructor(message: string) {
+        super(message)
+        this.name = 'ResponseError'
+    }
+
+    status: number
+}
+
+const exponentialBackoff: RequestInitWithRetry['retryDelay'] = (
+    attempt, error, response
+) => {
+    return Math.pow(2, attempt) * 1000
+}
+
+type retryOnRequestErrorFunc = (retries: number) => RequestInitWithRetry['retryOn']
+
+const retryOnRequestError: retryOnRequestErrorFunc = (retries) => {
+    return (attempt, error, response) => {
+        if (attempt >= retries) {
+            return false
+        }
+
+        if (response && response?.status < 500) {
```

Comment:
Should we handle for 404s here?

Reply:
handle them how? this is basically just making sure we only retry retryable errors (5xx)